### PR TITLE
B 1037 migrate create update project from hackathon

### PR DESCRIPTION
**Pull request by alexbeno**

### CODE REVIEW

File: src/pages/ProjectCreation/pages/ProjectInformations/components/ProjectLead/ProjectLead.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+  const contributors = data?.contributors;
+
+  const [selectedLead, setSelectedLead] = useState<components["schemas"]["ContributorSearchItemResponse"][]>([]);
+
+  useEffect(() => {
+    if (!selectedLead.length && contributors && value?.invited?.length) {
+      const findSelectedLead = value.invited
+        .map(invited => contributors?.find(lead => lead.githubUserId === invited))
+        .filter(l => l !== undefined);
```
Comment (haydencleary) [inline src/pages/ProjectCreation/pages/ProjectInformations/components/ProjectLead/ProjectLead.tsx:47]:
```suggestion
        .filter(l => l !== undefined);
```
```suggestion
        .filter(Boolean);
```

File: src/pages/ProjectCreation/pages/ProjectInformations/components/ProjectLead/ProjectLead.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    setQuery(query);
+  }, 500);
+
+  const onRemoveLead = (login: string) => {
+    setSelectedLead(selectedLead.filter(lead => lead.login !== login));
+  };
+
+  useEffect(() => {
+    if (onChange) {
+      onChange({ invited: selectedLead.map(lead => lead.githubUserId) });
```
Comment (haydencleary) [inline src/pages/ProjectCreation/pages/ProjectInformations/components/ProjectLead/ProjectLead.tsx:63]:
```suggestion
      onChange?.({ invited: selectedLead.map(lead => lead.githubUserId) });
```

File: src/pages/ProjectCreation/pages/ProjectInformations/components/ProjectLead/ProjectLead.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        />
+      )),
+    ],
+    [selectedLead, user]
+  );
+
+  return (
+    <div className=" flex w-full flex-col gap-2">
+      <FieldLabel id={githubUserId}>Project leads</FieldLabel>
+      <div className="flex flex-col gap-3 ">
```
Comment (haydencleary) [inline src/pages/ProjectCreation/pages/ProjectInformations/components/ProjectLead/ProjectLead.tsx:85]:
```suggestion
      <div className="flex flex-col gap-3">
```

File: src/pages/Projects/SubmitProject/index.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+export default function SubmitProject() {
+  const { T } = useIntl();
+  const { isLoggedIn } = useAuth();
+  const navigate = useNavigate();
+
+  const dispatchSession = useSessionDispatch();
+
+  const startProjectCreation = () => {
+    if (!isLoggedIn) {
```
Comment (haydencleary) [inline src/pages/Projects/SubmitProject/index.tsx:20]: Use truthy condition

File: src/pages/Projects/SubmitProject/index.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+  return (
+    <Card className="mb-4 flex h-fit flex-row gap-4 p-6">
+      <div className="flex-1 text-sm leading-4">{T("project.details.create.description")}</div>
+      <Button
+        htmlType="submit"
+        size={ButtonSize.Sm}
+        type={ButtonType.Primary}
+        width={Width.Fit}
+        onClick={() => startProjectCreation()}
```
Comment (haydencleary) [inline src/pages/Projects/SubmitProject/index.tsx:36]:
```suggestion
        onClick={startProjectCreation}
```

File: src/assets/img/index.ts
[the remark this answers is not in the document]
Comment (haydencleary) [reply src/assets/img/index.ts:1]: Maybe use individual named exports ?

File: src/components/New/Layout/Flex.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+}) => {
+  return (
+    <div
+      className={cn(
+        "flex",
+        `flex-${direction}`,
+        `justify-${justify}`,
+        `items-${item}`,
+        `flex-${wrap}`,
+        `gap-${gap}`,
```
[the remark this answers is not in the document]
Comment (kaelsky) [reply src/components/New/Layout/Flex.tsx:35]: @alexbeno what about using simple `Flex`, `Center`, `Box`, `Grid` elements by using tailwind `className`, just for semantic purpose ?

File: src/pages/ProjectCreation/pages/ProjectInformations/utils/ProjectInformations.utils.tsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import { OrganizationSessionStorageInterface } from "../../../commons/hooks/useProjectCreationSession";
+
+export const getSelectedRepoIds = (orgs: OrganizationSessionStorageInterface[]) => {
```
[the remark this answers is not in the document]
Comment (pixelfact) [reply src/pages/ProjectCreation/pages/ProjectInformations/utils/ProjectInformations.utils.tsx:3]: Done

**Review by kaelsky — APPROVED** [review]
LGTM !

**Pull request merged**